Releases.md

```diff
@@ -42,6 +43,16 @@ This strategy allows diagnostic tools such as `kata-runtime kata-env` to record
 
 Note that although hypervisor and guest kernel both have versions, these are not updated for new releases as they are not core components developed by the Kata community.
 
+### Component version compatibility
+
+Semantic versioning allows users to determine how a new version of a
+particular component relates to the previous version of that component.
+However, it is not possible to compare two *different* Kata component
```

Comment:
Can you explain or expand on why it is 'not possible to compare...' - it's not quite clear to me here what you are trying to explain.

Reply:
Nor me! :smile:

I think we're all struggling to really understand what this actually means, but I'm trying to find words, so if you can suggest some based on what follows, I'm all ears! ;)

It's rather difficult to write this. What I'm trying to state is that although we "use" semver for each component, since we require all components to have the same version to be compatible, we require our own... cough.. interpretation of what semver means. It could be argued that semver is implicitly only meant for comparing versions *of the same component*.

However since we're imposing the extra requirement that all components need to be upgraded in lock-step (all components having the same version), a gRPC protocol change (what would nominally be an "API breakage" for a single component) for example can be handled by a MINOR version number change only (since this detail is invisible to the user *assuming they upgrade all components in lock-step*).

For example, if the proxy is at version `1.2.0` but the other components are all at `1.1.0`, there are no guarantees the system will work.

I'll also probably add words based on @gnawux's comments to the ML (http://lists.katacontainers.io/pipermail/kata-dev/2018-June/000232.html) too.

Reply:
Sure, but 1.1.1 and 1.1.0 should be fine.  Right?

Reply:
@grahamwhaley, @egernst - should we just drop the "Component version compatibility" section entirely then do you think?

Reply:
Re-ping @grahamwhaley, @egernst.

Reply:
Thoughts guys...? :)

Reply:
Hi @jodh-intel - I'm prone to put this PR on hold, pretty much just due to this semver related section, until we have the Kata versioning and release flow defined - and then we can reference that.
I believe @jcvenegas is planning a first draft around releases, versioning and backport/lifetimes - let's wait for that.
As such, I'm going to mark this PR as WIP, just to make it clear it is pending an update.